### PR TITLE
python3Packages.pipetools: init at 1.1.0

### PR DESCRIPTION
**Pull request by RmiTtro**
## Description of changes

Added python lib: https://0101.github.io/pipetools/doc/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pipetools/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  ];
+  
+  pythonImportsCheck = [
+    "pipetools"
+  ];
+
+  meta = with lib; {
+    description = "A library that enables function composition similar to using Unix pipes";
+    homepage = https://0101.github.io/pipetools/;
+    license = licenses.mit;
```
Comment (DontEatOreo) [inline pkgs/development/python-modules/pipetools/default.nix:35]:
```suggestion
  meta = {
    description = "A library that enables function composition similar to using Unix pipes";
    homepage = https://0101.github.io/pipetools/;
    license = lib.licenses.mit;
```

File: pkgs/development/python-modules/pipetools/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, setuptools
+}:
```
Comment (DontEatOreo) [inline pkgs/development/python-modules/pipetools/default.nix:6]:
```suggestion
{
  lib,
  buildPythonPackage,
  fetchFromGitHub,
  pytestCheckHook,
  setuptools,
}:
```

File: pkgs/development/python-modules/pipetools/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchFromGitHub {
+    owner = "0101";
+    repo = pname;
+    rev = "6cba9fadab07a16fd85eed16d5cffc609f84c62b";
+    hash = "sha256-BoZFePQCQfz1dkct5p/WQLuXoNX3eLcnKf3Mf0fG6u8=";
+  };
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
```
Comment (DontEatOreo) [inline pkgs/development/python-modules/pipetools/default.nix:22]:
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```

File: pkgs/development/python-modules/pipetools/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  propagatedBuildInputs = [
+    setuptools
+  ];
```
Comment (DontEatOreo) [inline pkgs/development/python-modules/pipetools/default.nix:26]:
```suggestion
  propagatedBuildInputs = [ setuptools ];
```

File: pkgs/development/python-modules/pipetools/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  
+  pythonImportsCheck = [
+    "pipetools"
+  ];
```
Comment (DontEatOreo) [inline pkgs/development/python-modules/pipetools/default.nix:30]:
```suggestion
  pythonImportsCheck = [ "pipetools" ];
```

**Review by 4evy — CHANGES_REQUESTED** [review]

**Comment by DontEatOreo** [comment]
For the formatting changes please run `nix run nixpkgs#nixfmt-rfc-style -- pkgs/development/python-modules/pipetools/default.nix`

**Comment by RmiTtro** [comment]
> For the formatting changes please run `nix run nixpkgs#nixfmt-rfc-style -- pkgs/development/python-modules/pipetools/default.nix`

Will do, thank you for the quick review.

**Comment by RmiTtro** [comment]
Result of `nixpkgs-review pr 309716` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pipetools</li>
    <li>python311Packages.pipetools.dist</li>
    <li>python312Packages.pipetools</li>
    <li>python312Packages.pipetools.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pipetools/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  propagatedBuildInputs = [ setuptools ];
+
+  pythonImportsCheck = [ "pipetools" ];
+
+  meta = with lib; {
+    description = "A library that enables function composition similar to using Unix pipes";
+    homepage = "https://0101.github.io/pipetools/";
+    license = licenses.mit;
```
Comment (DontEatOreo) [inline pkgs/development/python-modules/pipetools/default.nix:30]:
```suggestion
  meta = {
    description = "A library that enables function composition similar to using Unix pipes";
    homepage = "https://0101.github.io/pipetools/";
    license = lib.licenses.mit;
```

**Comment by DontEatOreo** [comment]
Please rebase and squash 31ef449727d9e1ea4526d7df269f7ab485d891fb into a19e956b340d22dc27f64765dfec8736f9e9af14

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 309716` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pipetools</li>
    <li>python311Packages.pipetools.dist</li>
    <li>python312Packages.pipetools</li>
    <li>python312Packages.pipetools.dist</li>
  </ul>
</details>

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 309716` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pipetools</li>
    <li>python311Packages.pipetools.dist</li>
    <li>python312Packages.pipetools</li>
    <li>python312Packages.pipetools.dist</li>
  </ul>
</details>

**Review by 4evy — APPROVED** [review]

**Review by matthiasbeyer — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 309716` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pipetools</li>
    <li>python311Packages.pipetools.dist</li>
    <li>python312Packages.pipetools</li>
    <li>python312Packages.pipetools.dist</li>
  </ul>
</details>

**Pull request merged**